yang/ietf-acl-enh.yang

```diff
@@ -1,529 +1,634 @@
+     Relating to IETF Documents
+     (http://trustee.ietf.org/license-info).
+
+     This version of this YANG module is part of RFC XXXX; see
+     the RFC itself for full legal notices.";
+
+  revision 2022-10-24 {
+    description
+      "Initial revision.";
+    reference
+      "RFC XXXX: Extensions to the Access Control Lists (ACLs)
+                 YANG Model";
+  }
+
+  feature match-on-payload {
+    description
+      "Match based on a pattern is supported.";
+  }
+
+  identity offset-type {
+    description
+      "Base identity for payload offset type.";
+  }
+
+  identity layer3 {
+    base offset-type;
+    description
+      "IP header.";
+  }
+
+  identity layer4 {
+    base offset-type;
+    description
+      "Transport header (e.g., TCP or UDP).";
+  }
+
+  identity payload {
+    base offset-type;
+    description
+      "Transport payload. For example, this represents the beginning
+       of the TCP data right after any TCP options.";
+  }
+
+   identity tcp-flag {
+    description
+      "Base Identity for the TCP Flags.";
+  }
+
+  identity ACK {
+    base tcp-flag;
+    description
+      "The acknowledgment ACK TCP flag is used to acknowledge the 
+      successful receipt of a packet. As we can see from the diagram above, 
+      the receiver sends an ACK as well as a SYN in the second step
+      of the three-way handshake process to tell the sender 
+      that it received its initial packet.";
+  }
+
+  identity SYN {
+    base tcp-flag;
+    description
+      "The synchronization flag is used to establish a three-way handshake 
+      between two hosts. Only the first packet from both the sender and 
+      receiver should have this flag set. The following diagram 
+      illustrates a three-way handshake process.";
+  }
+
+  identity FIN {
+    base tcp-flag;
+    description
+      "The finished flag means there is no more data from the sender. 
+      Therefore, it is used in the last packet sent from the sender. 
+      It frees the reserved resources and gracefully terminates the connection";
+  }
+
+  identity URG {
+    base tcp-flag;
+    description
+      "he urgent flag is used to notify the receiver to process 
+      the urgent packets before processing all other packets. 
+      The receiver will be notified when all known urgent 
+      data has been received.";
+    reference
+     "RFC 6093";
+  }
+
+  identity PSH {
+    base tcp-flag;
+    description
+      "The push flag is similar to the URG flag and tells the 
+      receiver to process these packets as they are received 
+      instead of buffering them. ";
+  }
+
+  identity RST {
+    base tcp-flag;
+    description
+      "The reset flag gets sent from the receiver to the sender 
+      when a packet is sent to a particular host that was not expecting it.";
+  }
+
+  identity R1 {
+    base tcp-flag;
+    description
+      "The R1 retransmission threshold has been reached.";
```

Comment:
Not sure what retransmission is mentioned here. Please check the full flags at: https://www.iana.org/assignments/tcp-parameters/tcp-parameters.xhtml#tcp-header-flags